tests/e2e/performance/test_pvc_snapshot_performance.py

```diff
@@ -252,27 +301,186 @@ def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
+        sf_data["spec"]["workload"]["args"]["files"] = files
+        sf_data["spec"]["workload"]["args"]["threads"] = threads
+        sf_data["spec"]["workload"]["args"]["storageclass"] = storageclass
+
+        """
+        Calculating the size of the volume that need to be test, it should
+        be at least twice in the size then the size of the files, and at
+        least 100Gi.
+
+        Since the file_size is in Kb and the vol_size need to be in Gb, more
+        calculation is needed.
+        """
+        total_files = int(files * threads)
+        total_data = int(files * threads * file_size / constants.GB2KB)
+        data_set = int(total_data * 3)  # calculate data with replica
+        vol_size = data_set if data_set >= 100 else 100
+        sf_data["spec"]["workload"]["args"]["storagesize"] = f"{vol_size}Gi"
+
+        environment = get_environment_info()
+        if not environment["user"] == "":
+            sf_data["spec"]["test_user"] = environment["user"]
+        else:
+            # since full results object need this parameter, initialize it from CR file
+            environment["user"] = sf_data["spec"]["test_user"]
+
+        sf_data["spec"]["clustername"] = environment["clustername"]
+        log.debug(f"The smallfile yaml file is {sf_data}")
+
+        # Deploy the ripsaw operator
+        log.info("Apply Operator CRD")
+        ripsaw.apply_crd("resources/crds/ripsaw_v1alpha1_ripsaw_crd.yaml")
+
+        all_results = []
+
+        for test_num in range(self.tests_numbers):
+
+            # deploy the smallfile workload
+            log.info("Running SmallFile bench")
+            sf_obj = OCS(**sf_data)
+            sf_obj.create()
+
+            # wait for benchmark pods to get created - takes a while
+            for bench_pod in TimeoutSampler(
+                240, 10, get_pod_name_by_pattern, "smallfile-client",
+                constants.RIPSAW_NAMESPACE
+            ):
+                try:
+                    if bench_pod[0] is not None:
+                        small_file_client_pod = bench_pod[0]
+                        break
+                except IndexError:
+                    log.info("Bench pod not ready yet")
+
+            bench_pod = OCP(kind="pod", namespace=constants.RIPSAW_NAMESPACE)
+            log.info("Waiting for SmallFile benchmark to Run")
+            assert bench_pod.wait_for_resource(
+                condition=constants.STATUS_RUNNING,
+                resource_name=small_file_client_pod,
+                sleep=30,
+                timeout=600
+            )
+            for item in bench_pod.get()["items"][1]["spec"]["volumes"]:
+                if "persistentVolumeClaim" in item:
+                    pvc_name = item["persistentVolumeClaim"]["claimName"]
+                    break
+            log.info(f"Benchmark PVC name is : {pvc_name}")
+            # Creation of 4M files on CephFS can take a lot of time
+            timeout = 7200
```

Comment:
pls consider changing the timeout value based on the number of files

Reply:
There are more then one parameter (number of files) that influence the timeout : Number of files / File size / Block size / Volume type / Platform, and maybe more.
in most cases the test will not reach the timeout, and the purpose of the  workload is just to fill up the volume with files and the time it takes does not meter.